execute/observation.go

```diff
@@ -259,68 +261,177 @@ func buildCombinedReports(
 	return combinedReports
 }
 
-// regroup converts the previous outcome to the observation format.
-// TODO: use same format for Observation and Outcome.
-func regroup(commitData []exectypes.CommitData) exectypes.CommitObservations {
-	groupedCommits := make(exectypes.CommitObservations)
-	for _, report := range commitData {
-		if _, ok := groupedCommits[report.SourceChain]; !ok {
-			groupedCommits[report.SourceChain] = []exectypes.CommitData{}
-		}
-		groupedCommits[report.SourceChain] = append(groupedCommits[report.SourceChain], report)
-	}
-	return groupedCommits
-}
-
-func readAllMessages(
+func (p *Plugin) getObsWithoutTokenData(
 	ctx context.Context,
 	lggr logger.Logger,
-	ccipReader reader.CCIPReader,
 	commitData []exectypes.CommitData,
-) (exectypes.MessageObservations, exectypes.CommitObservations, map[cciptypes.Bytes32]time.Time) {
+	observation exectypes.Observation,
+) (exectypes.Observation, error) {
 	messageObs := make(exectypes.MessageObservations)
 	availableReports := make(exectypes.CommitObservations)
-	messageTimestamps := make(map[cciptypes.Bytes32]time.Time)
+	msgHashes := make(exectypes.MessageHashes)
 
-	commitObs := regroup(commitData)
+	sort.Slice(commitData, func(i, j int) bool {
+		return exectypes.CompareCommitData(commitData[i], commitData[j])
+	})
 
-	for srcChain, reports := range commitObs {
-		if len(reports) == 0 {
-			continue
+	stop := false
+	gasSum := uint64(0)
```

Comment:
I don't think this is a viable approach, gas limit is an EVM concept, and even if it isn't, we should not constrain ourselves to a particular amount in compute/gas here since we could split into multiple reports in the 3rd phase (e.g, Solana/Aptos)